### PR TITLE
html: fix parsing of Textarea.value

### PR DESCRIPTION
**Pull request by fscherf**
Signed-off-by: Florian Scherf <f.scherf@pengutronix.de>

### CODE REVIEW

File: lona/html/parsing.py [hunk 1]
```diff
@@ -161,7 +161,11 @@ def handle_data(self, data):
             if not text:
                 return
 
-        self._node.append(TextNode(text))
+        if self._node.tag_name == 'textarea':
+            self._node.value = text
```
Comment (maratori) [inline lona/html/parsing.py:165]: I'm afraid it is complicated than that. For example following should pass.
```py
assert HTML('<textarea>abc<br>xyz</textarea>')[0].value == 'abc<br>xyz'
assert HTML('<textarea>aaa <b>bbb</b> ccc</textarea>')[0].value == 'aaa <b>bbb</b> ccc'
```

**Review by maratori — CHANGES_REQUESTED** [review]

File: lona/html/parsing.py (hunk 1 above)
Comment (fscherf) [reply lona/html/parsing.py:165]: @maratori You are right. That's what the javascript does too

Comment (fscherf) [reply lona/html/parsing.py]: @maratori Fixed!

File: lona/html/parsing.py [hunk 2]
```diff
@@ -128,6 +153,15 @@ def handle_starttag(self, tag, attrs):
 
             node_attributes[name] = value
 
+        # textareas
+        if self._node.tag_name == 'textarea':
+            self._node.value = self._node.value + self.render_start_tag(
```
Comment (maratori) [inline lona/html/parsing.py:158]: It seems there is an easier solution. Just add
```py
class NodeHTMLParser(HTMLParser):
    CDATA_CONTENT_ELEMENTS = HTMLParser.CDATA_CONTENT_ELEMENTS + ("textarea",)
```

Comment (fscherf) [reply lona/html/parsing.py]: @maratori Oh! Thats much simpler indeed :D Thanks for researching this, this class attribute seems not to be documented

Fixed!

**Review by maratori — APPROVED** [review]

**Comment by fscherf** [comment]
@maratori Thanks for reviewing :)

**Pull request merged**